static/src/javascripts/projects/commercial/modules/messenger/viewport.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+import detect from 'lib/detect';
+import fastdom from 'lib/fastdom-promise';
+import { register } from 'commercial/modules/messenger';
+
+let w = window;
+let iframes = {};
+let iframeCounter = 0;
+let taskQueued = false;
+let lastViewport;
+
+const lastViewportRead = fastdom.read(() => {
+    lastViewport = detect.getViewport();
+});
+
+const reset = (window_: WindowProxy) => {
+    w = window_ || window;
+    taskQueued = false;
+    iframes = {};
+    iframeCounter = 0;
+};
+
+const sendViewportDimensions = function(iframeId) {
```

Comment:
Just checking: Is using `function(iframeId)` rather than an arrow function intentional? (I assume due to `this`?)